source/main.cpp

```diff
@@ -118,25 +147,26 @@ class SocketDemo {
     bool send_http_request()
     {
         /* loop until whole request sent */
-        const char buffer[] = R"(GET / HTTP/1.1
-Host: ifconfig.io
-Connection: close
-
-)";
+        const char buffer[] = "GET / HTTP/1.1\r\nHost: ifconfig.io\r\nConnection: close\r\n\r\n";
```

Comment:
Why not keeping the raw string literal ?

Reply:
for brevity, the message is immaterial to the example and the raw string literal breaks the indentation

Reply:
amended to have break lines